File: wallendar-app/app/src/main/java/com/td/wallendar/utils/mappers/GroupMapper.java
package com.td.wallendar.utils.mappers;

import com.td.wallendar.dtos.response.GroupResponse;
import com.td.wallendar.models.Group;

import java.util.ArrayList;
import java.util.List;

public class GroupMapper {

    public static List<Group> toModel(List<GroupResponse> groupsResponses) {
        List<Group> groups = new ArrayList<>();
        for (GroupResponse group : groupsResponses) {
            groups.add(toModel(group));
        }
        return groups;
    }

    public static Group toModel(GroupResponse groupResponse) {
        return new Group(
                groupResponse.getId(),
                groupResponse.getTitle(),
                ApplicationUserMapper.toModel(groupResponse.getOwner()),
                ApplicationUserMapper.toModel(groupResponse.getMembers()),
                ChargeMapper.toModel(groupResponse.getCharges()),
                DebtMapper.toModel(groupResponse.getDebts()),
                PaymentMapper.toModel(groupResponse.getPayments())
        );
    }

}

File: wallendar-app/app/src/main/java/com/td/wallendar/utils/mappers/GroupMapper.kt
package com.td.wallendar.utils.mappers

import com.td.wallendar.dtos.response.GroupResponse
import com.td.wallendar.models.Group
import java.util.*

object GroupMapper {
    fun toModel(groupsResponses: MutableList<GroupResponse>): MutableList<Group> {
        val groups: MutableList<Group> = ArrayList()
        for (group in groupsResponses) {
            groups.add(toModel(group))
        }
        return groups
    }

    fun toModel(groupResponse: GroupResponse): Group {
        return Group(
                groupResponse.id,
                groupResponse.title,
                ApplicationUserMapper.toModel(groupResponse.owner),
                ApplicationUserMapper.toModel(groupResponse.members),
                ChargeMapper.toModel(groupResponse.charges),
                DebtMapper.toModel(groupResponse.debts),
                PaymentMapper.toModel(groupResponse.payments)
        )
    }
}